sql/parse/parse.go

```diff
@@ -1620,6 +1625,32 @@ func convertCreateTable(ctx *sql.Context, c *sqlparser.DDL) (sql.Node, error) {
 		return plan.NewCreateTableSelect(sql.UnresolvedDatabase(c.Table.Qualifier.String()), c.Table.Name.String(), selectNode, tableSpec, plan.IfNotExistsOption(c.IfNotExists), plan.TempTableOption(c.Temporary)), nil
 	}
 
+	// Get the table's default collation
+	collation := sql.Collation_Default
+	if len(c.TableSpec.Options) > 0 {
```

Comment:
Why isn't this logic in TableSpecToSchema?

Reply:
TableSpecToSchema can return the collation in addition to a schema

Reply:
I've added a boolean to specify whether we want to force an invalid collation (that will be picked by analysis later) or parse the collation.